adoc/extensions/sycl_khr_static_addrspace_cast.adoc

```diff
@@ -0,0 +1,97 @@
+[[sec:khr-static-addrspace-cast]]
+= SYCL_KHR_STATIC_ADDRSPACE_CAST
+
+[code]#sycl::address_space_cast# does two things: first, it checks whether a
+given raw pointer can be cast to a specific address space; and second, it
+performs the requested cast.
+In cases where a developer is asserting that a raw pointer points to an object
+in a specific address space, run-time checks are not required and may have
+undesirable performance impact.
+This extension defines [code]#static_addrspace_cast# to provide developers a
+mechanism which casts with no run-time checks, enabling address space casts
+without any performance overhead.
+
+[[sec:khr-static-addrspace-cast-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-static-addrspace-cast-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_STATIC_ADDRSPACE_CAST# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+[[sec:khr-static-addrspace-cast-functions]]
+== Static address space cast functions
+
+[source,role=synopsis]
+----
+
+namespace sycl::khr {
+
+template <addrspace Space, typename ElementType>
+multi_ptr<ElementType, Space, access::decorated::no>
+static_addrspace_cast(ElementType* ptr);
+
+template <addrspace Space, typename ElementType, access::decorated DecorateAddress>
+multi_ptr<ElementType, Space, DecorateAddress>
+static_addrspace_cast(multi_ptr<ElementType, addrspace_generic, DecorateAddress> ptr);
+
+} // namespace khr::sycl
+----
+
+.[apidef]#static_addrspace_cast#
+[source,role=synopsis,id=api:khr-static-addrspace-cast-static_addrspace_cast]
+----
+template <addrspace Space, typename ElementType>
```

Comment:
```suggestion
template <access::address_space Space, typename ElementType>
```

Maybe we want this instead?  Note that the `addrspace` alias depends on #633.  Is #633 targeted for SYCL-Next?  If so, I think we want this KHR to work even for SYCL-2020.

Reply:
You're right; #633 is being targeted for SYCL-Next, so we should use the old enum here. Fixed in c5ccd25458c75c77e52e05dc6c466fe14a009099.